CHANGELOG.md

```diff
@@ -23,11 +25,24 @@
 - `az_iot_provisioning_client_init()` parameter `global_device_endpoint` renamed to `global_device_hostname`.
 - Renamed the macro `AZ_SPAN_NULL` to `AZ_SPAN_EMPTY`.
 - Renamed the `az_result` value `AZ_ERROR_INSUFFICIENT_SPAN_SIZE` to `AZ_ERROR_NOT_ENOUGH_SPACE`.
+- Made the helper macros `AZ_RETURN_IF_FAILED` and `AZ_RETURN_IF_NOT_ENOUGH_SIZE` internal.
+- Behavioral change to disallow passing `NULL` to `az_context` and update documentation.
+- Removed `AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_DUPLICATE` and `AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_RETAIN` from `az_iot_hub_client.h`.
 
 ### Bug Fixes
 
+- Fix the strict-aliasing issue in `_az_isfinite()`.
```

Comment:
As per `_az_` prefix, this is internal, do we need this in the changelog? If some public function was affected by this, I'd mention the stack aliasing issue, otherwise don't mention?